File: app/src/main/java/com/sanousun/mdweather/ui/activity/BaseActivity.java
package com.sanousun.mdweather.ui.activity;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;
import rx.subscriptions.CompositeSubscription;

public abstract class BaseActivity extends AppCompatActivity {

    protected abstract int getLayoutResID();

    protected abstract void initView();

    protected abstract void initData();

    protected abstract void initEvent();

    protected CompositeSubscription mCompositeSubscription;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutResID());
        mCompositeSubscription = new CompositeSubscription();
        ButterKnife.bind(this);
        initView();
        initData();
        initEvent();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mCompositeSubscription.clear();
    }
}

File: app/src/main/java/com/sanousun/mdweather/activity/BaseActivity.kt
package com.sanousun.mdweather.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import io.reactivex.disposables.CompositeDisposable

/**
 * Created by dashu on 2017/6/25.
 * 基类的Activity
 */

abstract class BaseActivity : AppCompatActivity() {

    val disposables: CompositeDisposable = CompositeDisposable()

    /**
     * 获取布局文件id
     */
    abstract fun setLayoutId(): Int

    /**
     * 初始化数据，比如说处理intent
     */
    abstract fun initData(savedInstanceState: Bundle?)

    /**
     * 初始化view
     */
    abstract fun initView()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(setLayoutId())
        initData(savedInstanceState)
        initView()
    }

    override fun onDestroy() {
        super.onDestroy()
        disposables.clear()
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item?.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
            else -> {
                return super.onOptionsItemSelected(item)
            }
        }
    }
}